app/services/make_an_offer.rb

```diff
@@ -87,15 +114,23 @@ def validate_course_data
     errors.add(:offered_course, "does not belong to provider #{current_provider.code}, it belongs to #{provider.code}") if providers_dont_match
   end
 
-  def validate_offer_conditions
-    return if @offer_conditions.blank?
+  def validate_further_conditions
+    return if further_conditions.blank?
 
-    unless @offer_conditions.is_a?(Array)
-      errors.add(:offer_conditions, 'must be an array')
-      return
+    further_conditions.each_with_index do |value, index|
+      errors.add("further_conditions#{index}", "must be #{MAX_CONDITION_LENGTH} characters or fewer") if value && value.length > MAX_CONDITION_LENGTH
     end
+  end
+
+  def validate_conditions_array
+    return if offer_conditions.is_a?(Array)
+
+    errors.add(:offer_conditions, 'must be an array')
```

Comment:
This isn't a useful error message for the user. How does it happen? Should it be an exception?

Reply:
Removed